File: backend/app/models/cluster.py
from pydantic import BaseModel
from typing import List, Dict, Any

from .player import PlayerStats


class PlayerCluster(BaseModel):
    cluster_id: int
    players: List[PlayerStats]
    centroid: Dict[str, float]


class ClusteringResult(BaseModel):
    clusters: List[PlayerCluster]
    season: str
    num_clusters: int

File: frontend/types/cluster.ts
import { PlayerStats } from './player';

export interface PlayerCluster {
  cluster_id: number;
  players: PlayerStats[];
  centroid: Record<string, number>;
}

export interface ClusteringResult {
  clusters: PlayerCluster[];
  season: string;
  num_clusters: number;
}

export interface ClusteringVisualizationNode {
  id: string;
  name: string;
  team: string;
  position: string;
  cluster: number;
  stats: PlayerStats;
  radius: number;
  x?: number;
  y?: number;
  fx?: number | null;
  fy?: number | null;
}

export interface ClusteringVisualizationLink {
  source: string | ClusteringVisualizationNode;
  target: string | ClusteringVisualizationNode;
  value: number;
}

export interface ClusteringVisualizationData {
  nodes: ClusteringVisualizationNode[];
  links: ClusteringVisualizationLink[];
}
